File: src/Domain/Entities/SubscriptionEntity.cs
using VibraHeka.Domain.Commerce.Enums;
using VibraHeka.Domain.Common.Enums;

namespace VibraHeka.Domain.Entities;

public class SubscriptionEntity : BaseAuditableEntity
{
    public string SubscriptionID { get; set; } = string.Empty;

    public string UserID { get; set; } = string.Empty;

    public DateTimeOffset StartDate { get; set; } = DateTimeOffset.UtcNow;

    public DateTimeOffset EndDate { get; set; } = DateTimeOffset.UtcNow;

    public string ExternalSubscriptionItemID { get; set; } = string.Empty;

    public string ExternalSubscriptionID { get; set; } = string.Empty;

    public string ExternalCustomerID { get; set; } = string.Empty;

    public string CheckoutSessionUrl { get; set; } = string.Empty;

    public DateTimeOffset CheckoutSessionExpiresAt { get; set; } = DateTimeOffset.UtcNow;

    public OrderStatus Status { get; set; } = OrderStatus.Draft;

    public SubscriptionStatus SubscriptionStatus { get; set; } = SubscriptionStatus.Created;

    public bool IsActive()
    {
        return SubscriptionStatus is SubscriptionStatus.Active or SubscriptionStatus.Trialing or SubscriptionStatus.ToBeCancelled;
    }
}

File: src/Infrastructure/terraform/Lambdas/Payments/src/Entities/SubscriptionEntity.ts
export default interface SubscriptionEntity {
    SubscriptionID : string;
    UserID: string;
    StartDate: Date;
    EndDate: Date;
    ExternalSubscriptionID: string;
    ExternalSubscriptionItemID: string;
    ExternalCustomerID: string;
    OrderType: string;
    Status: string;
    SubscriptionStatus: string;
    Created: Date;
    CreatedBy: string;
    LastModified: Date;
    LastModifiedBy: string;
}